src/merkle.cpp

```diff
@@ -47,7 +47,7 @@ namespace libtorrent
 		TORRENT_ASSERT(tree_node > 0);
 		// even numbers have their sibling to the left
 		// odd numbers have their sibling to the right
-		return tree_node + (tree_node&1?1:-1);
+		return tree_node + ((tree_node&1)?1:-1);
```

Comment:
I was worried that this was an actual error, but the operator precedence made the previous version work too (but I agree this is cleaner). If it had been an actual error the test would have to be fixed as well